src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -415,40 +426,148 @@ class spell_sha_elemental_blast : public SpellScriptLoader
     }
 };
 
-// 194084 - Flametongue
-class spell_sha_flametongue : public SpellScriptLoader
+// 318038 - Flametongue Weapon
+class spell_sha_flametongue_weapon : public SpellScript
+{
+    PrepareSpellScript(spell_sha_flametongue_weapon);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_FLAMETONGUE_WEAPON_ENCHANT });
+    }
+
+    bool Load() override
+    {
+        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
+    }
+
+    void HandleEffectHitTarget(SpellEffIndex /*effIndex*/)
+    {
+        Player* player = GetCaster()->ToPlayer();
+        uint8 slot = EQUIPMENT_SLOT_MAINHAND;
+        if (player->GetPrimarySpecialization() == TALENT_SPEC_SHAMAN_ENHANCEMENT)
+            slot = EQUIPMENT_SLOT_OFFHAND;
+
+        Item* targetItem = player->GetItemByPos(INVENTORY_SLOT_BAG_0, slot);
+        if (!targetItem || !targetItem->GetTemplate()->IsWeapon())
+            return;
+
+        player->CastSpell(targetItem, SPELL_SHAMAN_FLAMETONGUE_WEAPON_ENCHANT, true);
+    }
+
+    void Register()
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_sha_flametongue_weapon::HandleEffectHitTarget, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 319778  - Flametongue - SPELL_SHAMAN_FLAMETONGUE_WEAPON_AURA
+class spell_sha_flametongue_weapon_aura : public AuraScript
+{
+    PrepareAuraScript(spell_sha_flametongue_weapon_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_FLAMETONGUE_ATTACK });
+    }
+
+    void HandleEffectProc(AuraEffect* aurEff, ProcEventInfo& eventInfo)
+    {
+        PreventDefaultAction();
+
+        Unit* attacker = eventInfo.GetActor();
+        int32 damage = int32(attacker->GetTotalAttackPowerValue(BASE_ATTACK) * 0.0264f);
```

Comment:
where does 0.0264f come from?

Reply:
Tooltip. Most attack power formulas are in the tooltips

Reply:
It's also in Attack power coefficient on the spell

Reply:
It's also in Attack power coefficient on the spell

Reply:
@Warpten is it possible to get it through some object rather than hardcode then?
I've tried to use `aurEff->GetSpellEffectInfo()->BonusCoefficient` and `aurEff->GetSpellEffectInfo()->BonusCoefficientFromAP`, but both return 0.0f

Reply:
> It's also in Attack power coefficient on the spell

Not for this spell. I also was not able to get the coefficient value from anywhere.

Reply:
@Robingadko then again Spellwork gets it from somewhere, so there has to be someway to get it without hardcoding.

Reply:
@matanshukry I may be wrong but as far as I understand SpellWork takes that metadata info from SpellDescriptionVariables.db2 which is not loaded by server.

Reply:
@matanshukry According to tooltip the damage is max((0.0264 * Attack power), 1) so that there's no zero damage at low levels.

Reply:
@Robingadko I missed that max, thanks